docs/spec/spec.md

```diff
@@ -3274,3 +3345,276 @@ type Product record {
 ```
 
 >**Note:** If the reference resolver returns an entity of a different type than the entity being resolved, a runtime error will be returned to the router. For example, if the resolver returns a `User` for a `Product` entity, a runtime error will occur.
+
+## 11. Experimental Features
+
+This section includes the experimental features in the Ballerina GraphQL package. There _might be_ backward-incompatible changes to these features in future releases. Once a feature is stabilized, it will be graduated as a standard feature.
+
+### 11.1 DataLoader
+
+The Ballerina GraphQL module provides the capability to batch queries to datasources and efficiently cache the retrieved data through the `graphql.dataloader` submodule.
```

Comment:
`batch queries` has a different meaning in the GraphQL context. Shall we change the wording there? Maybe we can briefly explain/mention the n+1 problem and dataloader helps to solve that.

Reply:
Update `queries` here with `data fetching`